feat(nexus): Nexus

piwi3910 opened the pull request:
**Description**
<!--
Initial release of Sonatype nexus OSS
-->
⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

piwi3910 committed:
Initial Release

piwi3910 committed:
fix code prettier

piwi3910 committed:
fix prettier and readme bug

piwi3910 commented:
please advise why your check is claiming "Missing README.md" even that the file is there

stavros-k commented:
> please advise why your check is claiming "Missing README.md" even that the file is there

Can you point me where exactly is the `README.md` file? Because I can't see it.

piwi3910 committed:
Create README.md

Manually added readme file, as visual studio and git doesn't seem to upload it.

piwi3910 commented:
ok seem good now, not sure what was going on, file wasn't included in the PR for some reason, and git didn't want to push is again as it was already existing. So i've re-added the file direct on github and that worked.

stavros-k committed:
apply some formatting and try with rofs

stavros-k commented:
Looks good in general!

Few pointers for the future:
- README.md it's autogen, just have to be there (even emtpy). Might need to tweak CI to create it if it's not there.
- CHANGELOG.md its autogen too, afaik it's autocreated also, so you can skip it!
- Try to copy another app (preferable `jackett) from stable instead of incubator, so you start with a more "clean" looking values.yaml

stavros-k commented:
So, now to the app specific parts.
1. Does this app have a webui? If so, should the port be HTTP instead? Or is it using the same port for other functions (non-http)?
2. I think `EXTRA_JAVA_OPTIONS` is not a valid env var for this image. I only saw on their repo `INSTALL4J_ADD_VM_PARAMS`. Can you verify that the env is ok?

stavros-k committed:
it's already running as non-root

piwi3910 commented:
> So, now to the app specific parts.
> 
> 1. Does this app have a webui? If so, should the port be HTTP instead? Or is it using the same port for other functions (non-http)?
> 2. I think `EXTRA_JAVA_OPTIONS` is not a valid env var for this image. I only saw on their repo `INSTALL4J_ADD_VM_PARAMS`. Can you verify that the env is ok?

you are correct, this is an mistake from the copy of the template
the env var is correctly: INSTALL4J_ADD_VM_PARAMS

piwi3910 commented:
the port will be used for the admin webgui, but also for the api and the artifact functions, and container repos.

stavros-k committed:
update variable

stavros-k commented:
> the port will be used for the admin webgui, but also for the api and the artifact functions, and container repos.

In that case, it's probably best to leave it TCP, (which is the default, so the `protocol: TCP` can be ommited :)

stavros-k committed:
clean values

stavros-k commented:
@all-contributors please add @piwi3910  for code

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
